File: Server Side/StudyStipendCalcAPI/StudyStipendCalcAPI/DTOs/StudentDtos/StudentDto.cs
using System;
using System.ComponentModel;

namespace StudyStipendCalcAPI.DTOs.StudentDtos
{
    public enum SemesterType
    { Spring, Autumn }
    public class StudentDto
    {
        public int Id { get; set; }
        public string NeptunCode { get; set; }
        public string ModulCode { get; set; }
        public string ModulName { get; set; }
        public string TelephelyName { get; set; }
        public int Year { get; set; }
        /// <summary>
        /// Tankör
        /// </summary>
        /// <value>The student grop.</value>
        public string StudentGrop { get; set; }
        public int ActiveSemester { get; set; }
        public int PassiveSemester { get; set; }
       
        /// <summary>
        /// Ösztöndíjindex
        /// </summary>
        /// <value>The index of the credit.</value>
        public double CreditIndex { get; set; }
        public double EarnedCredit { get; set; }
        public string FinancialState { get; set; }
        public int AccceptedCredit { get; set; }
        public int AllSemesters { get; set; }


        /// <summary>
        /// Ösztöndíjmutató
        /// </summary>
        /// <value>The index of the stipend.</value>
        public double StipendIndex { get; set; }
        /// <summary>
        /// ÖCSI
        /// </summary>
        /// <value>The index of the group.</value>
        public double GroupIndex { get; set; }
        public double ExceptedCredit { get; set; }
        public string YearOfEnrollment { get; set; }
        public int FinishedSemester { get; set; }
        /// <summary>
        /// Ösztöndíj összege
        /// </summary>
        /// <value>The stipend ammount.</value>
        public int StipendAmmount { get; set; }
        public int Uid { get; set; }
    }
}

File: Client side/App/src/app/_models/Student/StudentDto.ts
//hallgatói adatokat kezelő object
export class StudentDto {
    constructor(
        public id: number,
        public neptuncode: string,
        public modulcode: string,
        public modulname: string,
        public telephely: string,
        //évfolyam
        public year: number,
        //tankör
        public studentgroup: string,
        public activesemester: number,
        public passivesemester: number,
        //ösztöndíj index
        public creditindex: number,
        //megszerzett kreditek
        public earnedcredit: number, 
        public financialstate: string,
        //elismert kreditek
        public acceptedcredit:number,
        public allsemester: number,
        //ösztöndíjmutató
        public stipendindex: number,
        //öcsi
        public groupindex: number,
        //elvárt kredit
        public exceptedcredit: number,
        //beiratkozás
        public YearOfEnrollment: string,
        public finishedsemester: number,
        //ösztöndíjösszeg
        public stipendamount:number
    )
    {}
}